src/core/internal/arrayop.d

```diff
@@ -302,6 +362,25 @@ template not(alias tmlp)
 {
     enum not(Args...) = !tmlp!Args;
 }
+/// find index for which pred!haystack[index] is true or return -1
```

Comment:
Done.
Note that this already was a ddoc comment and also it's just a private internal helper template with only 5 lines of implementation.